File: bench/type4/adversarial/cases/reduction_minmax_anyall/Sum.java
import java.util.Arrays;

class Sum {
    static int javaLoop(int[] xs) {
        int total = 0;
        for (int x : xs) {
            total = total + x;
        }
        return total;
    }

    static int javaReduce(int[] xs) {
        return Arrays.stream(xs).reduce(0, (total, x) -> total + x);
    }

    static int javaProduct(int[] xs) {
        return Arrays.stream(xs).reduce(1, (product, x) -> product * x);
    }
}

File: bench/type4/adversarial/cases/reduction_minmax_anyall/sum.js
function jsSumLoop(xs) {
  let total = 0;
  for (const x of xs) {
    total += x;
  }
  return total;
}

function jsSumReduce(xs) {
  return xs.reduce((total, x) => total + x, 0);
}

function jsWrongSeed(xs) {
  return xs.reduce((total, x) => total + x, 1);
}

function jsProductReduce(xs) {
  return xs.reduce((product, x) => product * x, 1);
}
